### PR TITLE
Added functionality to connect increase of stats to level, stat reset…

### PR DESCRIPTION
**Pull request by kenneth-bakke**
… on class change

### CODE REVIEW

File: src/Character/Character.jsx [hunk 1]
```diff
@@ -43,10 +86,43 @@ export default function Character() {
...
+    for (let attr in newAttributes) {
+      updatedAttributes[attr] = newAttributes[attr];
+    }
+
+    setCharacterClass(newClassType);
+    setLevel(classStats.stats.level);
+    setAttributes(updatedAttributes);
+  };
+
+  const updateCategory = (e) => {
```
Comment (joseph-bakke) [inline src/Character/Character.jsx:113]: uwu what's this?

File: src/Character/Attribute.jsx [hunk 2]
```diff
@@ -33,14 +34,19 @@ export default function Attribute({ attribute, value }) {
 
   const updateAttribute = (e) => {
     e.preventDefault();
+    const diff = nextVal - val;
+    const nextLevel = level + diff;
+    console.log(nextLevel);
```
Comment (joseph-bakke) [inline src/Character/Attribute.jsx]: Nit: Extra console log

File: src/Character/Attribute.jsx [hunk 3]
```diff
@@ -5,7 +5,8 @@ import CharacterContext from './CharacterContext';
 export default function Attribute({ attribute, value }) {
   const [editAttribute, setEditAttribute] = useState(false);
   const [val, setVal] = useState(value);
```
Comment (joseph-bakke) [inline src/Character/Attribute.jsx]: I'm having a little bit of a hard time parsing what `val` and `nextVal` are supposed to represent here. Is there a different variable name you could choose to make it a little bit clearer?

File: src/Character/Character.jsx [hunk 4]
```diff
@@ -43,10 +86,43 @@ export default function Character() {
     return attributeList;
   };
 
+  const renderClassOptions = () => {
+    const classList = [];
+    const classNames = Object.keys(classes);
+    for (let cName of classNames) {
```
Comment (joseph-bakke) [inline src/Character/Character.jsx]: Nit: This is maybe my personal style preference, but I would do a `map` and spell out `className`.

`const classList = classNames.map((className) => <option key={sha256(className)}>{capitalize(className)}</option>);`

File: src/Character/Character.jsx [hunk 5]
```diff
@@ -43,10 +86,43 @@ export default function Character() {
...
+    setEditMode(!editMode);
+  };
+
+  const updateClass = (e) => {
+    const newClassType = e.target.value;
+    const classStats = classes[newClassType.toLowerCase()];
+    const newAttributes = classStats.stats.attributes;
+    const updatedAttributes = {};
+
+    for (let attr in newAttributes) {
```
Comment (joseph-bakke) [inline src/Character/Character.jsx]: Can you not just assign this directly by coping the object?

```
const classAtributes = { ...classStats?.stats?.attributes };

setAttributes(classAttributers);
```

**Comment by joseph-bakke** [comment]
Just a note. It's generally helpful to put a brief description in the start of the MR to explain what the changes are to help set context for the reviewer(s)

File: src/Character/Character.jsx (hunk 1 above)
Comment (kenneth-bakke) [reply src/Character/Character.jsx:113]: lmao removed

File: src/Character/Character.jsx (hunk 5 above)
Comment (kenneth-bakke) [reply src/Character/Character.jsx]: Totally works, and much cleaner looking.

File: src/Character/Character.jsx [hunk 6]
```diff
@@ -89,12 +89,10 @@ export default function Character() {
   };
 
   const renderClassOptions = () => {
-    const classList = [];
     const classNames = Object.keys(classes);
-    for (let cName of classNames) {
-      classList.push(<option key={sha256(cName)}>{capitalize(cName)}</option>);
-    }
-
+    const classList = classNames.map((className) => (
```
Comment (joseph-bakke) [inline src/Character/Character.jsx]: I realize now that I'm looking at this a second time that `className` is _sort of_ a keyword. It means something very specific in React (the actual class that will be applied to a DOM Element).

Should maybe consider renaming this to make it clear that this is specific to the Elden Ring class.

Comment (kenneth-bakke) [reply src/Character/Character.jsx]: Ooh good catch

Comment (kenneth-bakke) [reply src/Character/Character.jsx]: class too cuz of Class(), maybe classLabel? classType?

Comment (joseph-bakke) [reply src/Character/Character.jsx]: `classType` seems fine to me!

**Pull request merged**